pkg/pubsub/pubsub_init.go

```diff
@@ -0,0 +1,87 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+
+	psClient "github.com/open-policy-agent/gatekeeper/pkg/pubsub/client"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+func AddToManager(m manager.Manager) error {
+	r := newReconciler(m)
+	return add(m, r)
+}
+
+type Reconciler struct {
+	client.Client
+	*runtime.Scheme
+}
+
+func newReconciler(mgr manager.Manager) *Reconciler {
+	return &Reconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	c, err := controller.New("pusbub-config-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Secret{}},
+		&handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(
+		&source.Kind{Type: &corev1.ConfigMap{}},
+		&handler.EnqueueRequestForObject{})
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	providers := psClient.Providers()
+	if len(providers) == 0 {
+		log.Info("No pub sub tool is enabled")
+		return reconcile.Result{}, nil
+	}
+
+	if systems == nil {
+		log.Info("No pubsub system is requested")
+		return reconcile.Result{}, nil
+	}
+
+	if newConnection, ok := providers[request.Name]; ok {
+		config := &corev1.ConfigMap{}
```

Comment:
Are we sure we want to use config maps for this? It may be easier for driver authors to write their own CRD/controller.